### DISCUSSION TITLE
Dynamically adding rules/rule groups based on an input's value

### DISCUSSION THREAD

**Discussion by nheng1** [react-querybuilder/react-querybuilder#459] [Q&A]
Is it possible to dynamically add rules/rule groups based an input's value? For example: if I entered the first name of _Steve_, I would like to dynamically display another rule/group.

**Comment by jakeboone02** [comment]
Can you describe the intended user experience in more detail? Like, when should the new rule show up? When the user chooses "First Name" as the field of a rule, or when they enter something in the value editor, or only when they finish entering "Steve" in the value editor? Would the new rule/group need to be an immediate sibling of the "Steve" rule, or can it be appended as the last rule in the same group?

There are a lot of different ways you could go about this depending on these answers and more.

**Reply by nheng1** [reply]
Thanks for the quick reply!

This would be only when the user finishes entering "Steve" and we were looking for the new rule to be added as an immediate sibling to the "Steve" rule.

**Comment by nheng1** [comment]
Side-question: Wasn't sure if I should make a new topic or continue with this one, but I also had another question.

Is it possible to use the _ruleProcessor_ in formatQuery with the format of _sql_? I created a very simple rule processor and just let it log a console log to make sure it was working. When I used the _sql_ as the format, it would work as if it wasn't aware of the ruleProcessor. As a test, I used _mongodb_ as another format and then it was able to hit the ruleProcessor.

Thanks!

Example code:

```
import { formatQuery, defaultValueProcessorByRule } from 'react-querybuilder';
import 'react-querybuilder/dist/query-builder.scss';

const ReactQueryBuilder = () => {
    const query = {
        combinator: 'and',
        not: false,
        rules: [
            {
                field: 'firstName',
                operator: '=',
                value: 'S',
            },
            {
                field: 'lastName',
                operator: '=',
                value: 'Vai',
            },
        ],
    };

    const customRuleProcessor = (rule, options) => {
        console.log(rule, options);
        return defaultValueProcessorByRule(rule, options);
    };

    return (
        <div>
                {formatQuery(query, { format: 'sql', ruleProcessor: customRuleProcessor })}
        </div>
    )
};

export default ReactQueryBuilder;
```
